src/databricks/labs/ucx/workspace_access/manager.py

```diff
@@ -144,6 +148,51 @@ def apply_group_permissions(self, migration_state: MigrationState) -> bool:
         logger.info("Permissions were applied")
         return True
 
+    def apply_group_permissions_experimental(self, migration_state: MigrationState) -> bool:
+        if len(migration_state) == 0:
+            logger.info("No valid groups selected, nothing to do.")
+            return True
+        logger.info(
+            f"Applying the permissions to account groups. "
+            f"Total groups to apply permissions: {len(migration_state)}."
+        )
+        errors: list[Exception] = []
+        for migrated_group in migration_state.groups:
+            errors.extend(self._migrate_group_permissions_paginated(migrated_group))
+        if len(errors) > 0:
+            logger.error(f"Detected {len(errors)} errors while applying permissions")
+            raise ManyError(errors)
```

Comment:
```suggestion
        items = 0
        for migrated_group in migration_state.groups:
            items += self._migrate_group_permissions_paginated(migrated_group)
        logger.info(f"Migrated {items} items")
```
immediate failure in a resumable job is always better. we can add `@retried` on the call if needed.